ext/pcntl/pcntl.c

```diff
@@ -140,9 +140,17 @@ PHP_MSHUTDOWN_FUNCTION(pcntl)
 PHP_RSHUTDOWN_FUNCTION(pcntl)
 {
 	struct php_pcntl_pending_signal *sig;
+	int signo;
+	zval *handle;
+
+	/* Remove all registered signal handlers */
+	for (signo = 1; signo < PCNTL_G(num_signals); ++signo) {
+		handle = zend_hash_index_find(&PCNTL_G(php_signal_table), signo);
+		if (handle != NULL && Z_TYPE_P(handle) != IS_LONG) {
```

Comment:
This will skip SIG_IGN. This is not a big issue, but for consistency we could reset those as well.

Reply:
The behavior feels easier to explain if all signals are reset to their default disposition at the end of the request.

Reply:
Makes sense. Done.